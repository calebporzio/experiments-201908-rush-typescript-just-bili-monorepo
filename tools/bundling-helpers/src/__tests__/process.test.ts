/* eslint-env jest */

import { getProcessCliArgValuePair, MaybeNodeProcessContext } from '..'

describe('getProcessCliArgValuePair', () => {
  it('Snapshot', () => {
    const processFixture: MaybeNodeProcessContext = {
      argv: [
        '--foo',
        'bar',
        '--buz',
        '--config',
        '/path/to/fictious/tools/bundling-helpers/jest.config.js',
      ],
    }
    const subject = getProcessCliArgValuePair('--config', processFixture)
    expect(subject).toMatchSnapshot()
  })
  it('Missing CLI arguments returns null', () => {
    const subject = getProcessCliArgValuePair('--config')
    expect(subject).toBe(null)
  })
  it('Empty CLI arguments returns null', () => {
    // @ts-ignore
    const subject = getProcessCliArgValuePair('--config', { argv: [] })
    expect(subject).toBe(null)
  })
})
