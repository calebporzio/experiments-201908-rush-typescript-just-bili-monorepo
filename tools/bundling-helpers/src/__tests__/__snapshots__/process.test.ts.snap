// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProcessCliArgValuePair Snapshot 1`] = `
Array [
  "--config",
  "/path/to/fictious/tools/bundling-helpers/jest.config.js",
]
`;
